Exception about missing maven classes when opening the dev-ui (gradle based project)

### Describe the bug

When you generate the most simple quarkus app using Gradle:
https://code.quarkus.io/?b=GRADLE

Start it in dev mode with `./gradlew quarkusDev`

Open the dev ui http://localhost:8080/q/dev-ui/


### Expected behavior

No Exception, everything is working

### Actual behavior

I got different errors about missing Maven classes:

```
2024-12-13 10:05:26,851 ERROR [io.qua.dev.run.jso.JsonRpcCodec] (ForkJoinPool.commonPool-worker-9) Error in JsonRPC Call: java.util.concurrent.CompletionException: java.lang.NoClassDefFoundError: org/apache/maven/settings/building/SettingsBuildingException
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: java.lang.NoClassDefFoundError: org/apache/maven/settings/building/SettingsBuildingException
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:90)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver$Builder.build(MavenArtifactResolver.java:74)
        at io.quarkus.devtools.project.buildfile.MavenProjectBuildFile.getMavenResolver(MavenProjectBuildFile.java:151)
        at io.quarkus.devtools.project.buildfile.MavenProjectBuildFile.getProject(MavenProjectBuildFile.java:54)
        at io.quarkus.devtools.project.QuarkusProjectHelper.getCachedProject(QuarkusProjectHelper.java:59)
        at io.quarkus.devui.deployment.menu.ExtensionsProcessor.getQuarkusProject(ExtensionsProcessor.java:202)
        at io.quarkus.devui.deployment.menu.ExtensionsProcessor.lambda$getInstalledNamespaces$4(ExtensionsProcessor.java:124)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
        ... 6 more
Caused by: java.lang.ClassNotFoundException: org.apache.maven.settings.building.SettingsBuildingException
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        ... 14 more
```

Or:

```
2024-12-13 09:18:42,853 ERROR [io.qua.dev.run.jso.JsonRpcCodec] (ForkJoinPool.commonPool-worker-15) Error in JsonRPC Call: java.util.concurrent.CompletionException: java.lang.NoClassDefFoundError: org/apache/maven/model/profile/activation/ProfileActivator
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: java.lang.NoClassDefFoundError: org/apache/maven/model/profile/activation/ProfileActivator
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:90)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver$Builder.build(MavenArtifactResolver.java:74)
        at io.quarkus.devtools.project.buildfile.MavenProjectBuildFile.getMavenResolver(MavenProjectBuildFile.java:151)
        at io.quarkus.devtools.project.buildfile.MavenProjectBuildFile.getProject(MavenProjectBuildFile.java:54)
        at io.quarkus.devtools.project.QuarkusProjectHelper.getCachedProject(QuarkusProjectHelper.java:59)
        at io.quarkus.devui.deployment.menu.ExtensionsProcessor.getQuarkusProject(ExtensionsProcessor.java:202)
        at io.quarkus.devui.deployment.menu.ExtensionsProcessor.lambda$getInstalledNamespaces$4(ExtensionsProcessor.java:124)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
        ... 6 more
Caused by: java.lang.ClassNotFoundException: org.apache.maven.model.profile.activation.ProfileActivator
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        ... 14 more
```

Screenshot of the dev ui:

<img width="1098" alt="Image" src="https://github.com/user-attachments/assets/581771b6-cdb6-499c-bfc7-b788b7b464cd" />


### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.17.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java
@@ -448,7 +448,6 @@ public abstract class QuarkusDev extends QuarkusTask {
         if (getModules().isPresent() && !getModules().get().isEmpty()) {
             builder.addModules(getModules().get());
         }
-
         for (Map.Entry<String, ?> e : project.getProperties().entrySet()) {
             if (e.getValue() instanceof String) {
                 builder.buildSystemProperty(e.getKey(), e.getValue().toString());
@@ -465,11 +464,14 @@ public abstract class QuarkusDev extends QuarkusTask {
         builder.extensionDevModeConfig(appModel.getExtensionDevModeConfig())
                 .extensionDevModeJvmOptionFilter(extensionJvmOptions);
 
+        builder.jvmArgs("-Dgradle.project.path="
+                + getProject().getLayout().getProjectDirectory().getAsFile().getAbsolutePath());
+
         analyticsService.sendAnalytics(
                 DEV_MODE,
                 appModel,
                 Map.of(GRADLE_VERSION, getProject().getGradle().getGradleVersion()),
-                getProject().getBuildDir().getAbsoluteFile());
+                getProject().getLayout().getBuildDirectory().getAsFile().get());
 
         final Set<ArtifactKey> projectDependencies = new HashSet<>();
         for (ResolvedDependency localDep : DependenciesFilter.getReloadableModules(appModel)) {
@@ -562,6 +564,7 @@ public abstract class QuarkusDev extends QuarkusTask {
                         configuration.setCanBeConsumed(false);
                         configuration.extendsFrom(platformConfig);
                         configuration.getDependencies().add(getQuarkusGradleBootstrapResolver());
+                        configuration.getDependencies().add(getQuarkusMavenBootstrapResolver());
                         configuration.getDependencies().add(getQuarkusCoreDeployment(appModel));
                     });
             devModeDependencyConfiguration = getProject().getConfigurations()
@@ -585,7 +588,15 @@ public abstract class QuarkusDev extends QuarkusTask {
     }
 
     private Dependency getQuarkusGradleBootstrapResolver() {
-        final String pomPropsPath = "META-INF/maven/io.quarkus/quarkus-bootstrap-gradle-resolver/pom.properties";
+        return getQuarkusBootstrapResolver("quarkus-bootstrap-gradle-resolver");
+    }
+
+    private Dependency getQuarkusMavenBootstrapResolver() {
+        return getQuarkusBootstrapResolver("quarkus-bootstrap-maven-resolver");
+    }
+
+    private Dependency getQuarkusBootstrapResolver(String artifactId) {
+        final String pomPropsPath = "META-INF/maven/io.quarkus/" + artifactId + "/pom.properties";
         final InputStream devModePomPropsIs = DevModeMain.class.getClassLoader().getResourceAsStream(pomPropsPath);
         if (devModePomPropsIs == null) {
             throw new GradleException("Failed to locate " + pomPropsPath + " on the classpath");
@@ -608,9 +619,8 @@ public abstract class QuarkusDev extends QuarkusTask {
         if (devModeVersion == null) {
             throw new GradleException("Classpath resource " + pomPropsPath + " is missing version");
         }
-        Dependency gradleResolverDep = getProject().getDependencies()
+        return getProject().getDependencies()
                 .create(String.format("%s:%s:%s", devModeGroupId, devModeArtifactId, devModeVersion));
-        return gradleResolverDep;
     }
 
     private Dependency getQuarkusCoreDeployment(ApplicationModel appModel) {

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/ExtensionsProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/ExtensionsProcessor.java
@@ -63,7 +63,6 @@ public class ExtensionsProcessor {
                 && launchModeBuildItem.getDevModeType().get().equals(DevModeType.LOCAL)) {
 
             BuildTimeActionBuildItem buildTimeActions = new BuildTimeActionBuildItem(NAMESPACE);
-
             getCategories(buildTimeActions);
             getInstallableExtensions(buildTimeActions);
             getInstalledNamespaces(buildTimeActions);
@@ -145,6 +144,8 @@ public class ExtensionsProcessor {
                     }
 
                     return null;
+                } catch (IllegalStateException e) {
+                    return null;
                 } catch (QuarkusCommandException e) {
                     throw new RuntimeException(e);
                 }
@@ -198,7 +199,13 @@ public class ExtensionsProcessor {
     }
 
     private QuarkusProject getQuarkusProject() {
-        Path projectRoot = Paths.get(System.getProperty("user.dir")).toAbsolutePath().normalize();
+        Path projectRoot;
+        String gradlePath = System.getProperty("gradle.project.path");
+        if (gradlePath != null) {
+            projectRoot = Path.of(gradlePath);
+        } else {
+            projectRoot = Paths.get(System.getProperty("user.dir")).toAbsolutePath().normalize();
+        }
         return QuarkusProjectHelper.getCachedProject(projectRoot);
     }
 

--- a/extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-continuous-testing.js
+++ b/extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-continuous-testing.js
@@ -150,7 +150,9 @@ export class QwcContinuousTesting extends QwcHotReloadElement {
 
     _cancelObservers(){
         this._streamStateObserver.cancel();
-        this._streamResultsObserver.cancel();
+        if(this._streamResultsObserver){
+            this._streamResultsObserver.cancel();
+        }
     }
 
     _lastKnownState(){

--- a/extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-extensions.js
+++ b/extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-extensions.js
@@ -86,7 +86,8 @@ export class QwcExtensions extends observeState(LitElement) {
         _favourites: {state: true},
         _addDialogOpened: {state: true},
         _installedExtensions: {state: true, type: Array},
-        _selectedFilters: {state: true, type: Array}
+        _selectedFilters: {state: true, type: Array},
+        _addExtensionsEnabled: {state: true, type: Boolean},
     }
 
     constructor() {
@@ -96,12 +97,18 @@ export class QwcExtensions extends observeState(LitElement) {
         this._installedExtensions = [];
         this._filteritems = ["Favorites","Active","Inactive"];
         this._selectedFilters = this._getStoredFilters();
+        this._addExtensionsEnabled = false;
     }
 
     connectedCallback() {
         super.connectedCallback();
         this.jsonRpc.getInstalledNamespaces().then(jsonRpcResponse => {
-            this._installedExtensions = jsonRpcResponse.result;
+            if (jsonRpcResponse.result) {
+                this._installedExtensions = jsonRpcResponse.result;
+                this._addExtensionsEnabled = true;
+            }
+        }).catch(e => {
+            notifier.showErrorMessage("Could not list namespaces "+ e?.error?.message);
         });
     }
 
@@ -355,30 +362,32 @@ export class QwcExtensions extends observeState(LitElement) {
     }
 
     _renderAddDialog(){
-        return html`
-            <vaadin-dialog
-              theme="no-padding"
-              resizable
-              draggable
-              header-title="Add extension"
-              .opened="${this._addDialogOpened}"
-              @opened-changed="${(event) => {
-                this._addDialogOpened = event.detail.value;
-              }}"
-              ${dialogHeaderRenderer(
-                  () => html`
-                    <vaadin-button theme="tertiary" @click="${() => (this._addDialogOpened = false)}">
-                        <vaadin-icon icon="font-awesome-solid:xmark"></vaadin-icon>
-                    </vaadin-button>
-                  `,
-                  []
+        if (this._addExtensionsEnabled) {
+            return html`
+                <vaadin-dialog
+                theme="no-padding"
+                resizable
+                draggable
+                header-title="Add extension"
+                .opened="${this._addDialogOpened}"
+                @opened-changed="${(event) => {
+                    this._addDialogOpened = event.detail.value;
+                }}"
+                ${dialogHeaderRenderer(
+                    () => html`
+                        <vaadin-button theme="tertiary" @click="${() => (this._addDialogOpened = false)}">
+                            <vaadin-icon icon="font-awesome-solid:xmark"></vaadin-icon>
+                        </vaadin-button>
+                    `,
+                    []
+                    )}
+                ${dialogRenderer(
+                    () => html`<qwc-extension-add @inprogress="${this._installRequest}"></qwc-extension-add>`
                 )}
-              ${dialogRenderer(
-                () => html`<qwc-extension-add @inprogress="${this._installRequest}"></qwc-extension-add>`
-              )}
-            ></vaadin-dialog>
-            ${this._renderAddExtensionButton()}
-          `;
+                ></vaadin-dialog>
+                ${this._renderAddExtensionButton()}
+            `;
+        }
     }
     
     _renderAddExtensionButton(){

